Adds new item "go to definition" for drop-down menus of custom block callers

TheLogFather opened the pull request:
Really missed having this! -And so simple to add, too...

Not sure about the best text for it. Currently used "go to definition", but maybe "show definition"?
(It's a pity that, in English, the word 'definition' seems so different to the word 'define', as used on the block definition - makes the link potentially not quite so clear.)

I guess translations will need adding for it...

Commit:
Adds new item "go to definition" for drop-down menus of custom block callers.

griffpatch commented:
You the man :) please please scratch team take these awesome changes!
On 11 Jun 2015 14:04, "Aka DadOfMrLog" <notifications@github.com> wrote:

> Really missed having this! -And so simple to add, too...
>
> Not sure about the best text for it. Currently used "go to definition",
> but maybe "show definition"?
> (It's a pity that, in English, the word 'definition' seems so different to
> the word 'define', as used on the block definition - makes the link
> potentially not quite so clear.)
>
> I guess translations will need adding for it...
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/LLK/scratch-flash/pull/797
> Commit Summary
>
>    - Adds new item "go to definition" for drop-down menus of custom block
>    callers.
>
> File Changes
>
>    - *M* src/scratch/BlockMenus.as
>    <https://github.com/LLK/scratch-flash/pull/797/files#diff-0> (19)
>
> Patch Links:
>
>    - https://github.com/LLK/scratch-flash/pull/797.patch
>    - https://github.com/LLK/scratch-flash/pull/797.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LLK/scratch-flash/pull/797>.
>

ntlrsk commented:
I see this could be very helpful. For wording how about "see the define script" or "go to define block"? (Since "show" usually means on the Stage).

sclements commented on src/scratch/BlockMenus.as:
```diff
@@ -533,9 +533,28 @@ public class BlockMenus implements DragClient {
 		m.addItem('edit', editProcSpec);
+		if (block.op==Specs.CALL) {
+			/// TLF: alternative text? -maybe "show definition"...?
+			m.addItem('go to definition', gotoProcSpec);
```
Can you please change this to 'see the define script'?

TheLogFather replied:
Yup, I'll do that a bit later today - that 'define script' was the wording I'd come to prefer myself. :)
I did wonder about whether to merge this with #861 (the block highlighting PR) in some way - maybe, along with the "highlight callers" item (don't really like 'callers' that much...), having a "highlight define script" item, which actually also jumps to the script like this.
But I guess that PR is a considerably larger change, and this one does stand better by itself, so...

TheLogFather replied:
OK, I made the text change, plus a couple of minor tweaks.
My only comment after trying it out as "see the define script" would be that it makes the menu seem a bit wide compared to others (might some translations make it even wider?), and a tad unusual since other menu items (and tooltips) don't really use articles, i.e. "the", "a", etc...
Shall we start up SUREMI (Society for the Use of Real English in Menu Items)... ;)

Commit:
Change menu item text to "see the define script".
Change function name to match spec name (Spec.PROCEDURE_DEF="procDef").
Slight tidy-up of code.

ealgase commented:
This seems simple, but it seems that it would be very laggy.

TheLogFather commented:
@ealgase: why would you think that's the case...?

sclements commented:
Okay, looks like we're changing our mind about the text. Could you please make it say, "define"?

TheLogFather commented:
Just the single word "define"...?
I guess that makes a degree of sense - it's sorta like it's a verb (as with many other items), so you're telling the editor you want to 'define' this block... kinda... bit subtle, though...

ntlrsk commented:
@TheLogFather Yes, we were worrying about translation getting long, so the editor team ended up thinking this would be the most direct wording (even if a little subtle).

TheLogFather commented:
TBH, I think quite often it's not worth worrying overly much about these things, since we so often find things out by experience rather than by actually understanding the meaning - i.e. once we *see* what something does when we try it, we understand how to use it (and may even largely ignore the meaning behind the words themselves - just consider the ubiquitous "File" menu...)
Anyway, should be ready to push in a mo...

Commit:
Change menu item to just "define".
